_posts/2018-01-05-python101.md

```diff
@@ -0,0 +1,734 @@
+
+Now let’s look at a different kind of loop, called a ``while`` loop.
+
+## ``while`` loops
+While loops are also used for repeating blocks of code, but not for a fixed number of repetitions. The block will be run until some condition is met. Let’s consider a basic while loop:
+
+```python
+i = 1
+while i < 5:
+  print(i)
+  i += 1
+```
+Output:
+```
+<results>
+```
+
+Let’s break this down line by line. On the first line, we’ve created a counter variable i and set it to 1. The second line is the beginning of our loop: we’re going to keep running it until the condition ``i < 5`` is met. The next two indented lines are the block that will repeat. The first indented line is just a print statement, but the second one is more interesting: we’re incrementing the counter i by 1. i += 1 is just a more compact way of writing i = i+1, and it makes sure that we don’t endlessly get stuck in the loop: if we never increased i, the condition i<5 would always be true and we’d have an infinite loop which would never end. This is not something you usually want in your code. What about the condition? What if we changed it to i <= 5? We’d print the numbers from 1 to 5 this time, instead of 1 to 4.
+
+There are a bunch of numerical examples we could go over, but while loops get more interesting when we include user input. Let’s go over the code for a simple number guessing game:
+
+```python
+guess = 0
+while guess != to_be_guessed:
+  guess = int(input("New number: "))
+  if guess > 0:
+    if guess > to_be_guessed:
+      print("Number too large")
+    elif guess < to_be_guessed:
+      print("Number too small")
+  else:
+    print("Sorry that you're giving up!")
+    break
+
+if guess == to_be_guessed:
+  print("Congratulations. You made it!")
+```
+
+Let’s assume that to_be_guessed has been set to some random number outside of this block of code. The first line initializes a guess variable, and from there on it’s the while loop. The loop runs as long as the guess is incorrect. Inside the loop, the first thing we do is get a guess from the user - the input function takes a string using the prompt in quotes ("New number: "). The first if/else block checks to see if the guess is valid. We’ll talk later about other, more comprehensive ways this can be done. Inside the first if block we have another if/else block to tell the user how they did if they guessed over or under — note there’s no else block to handle if they guessed correctly. What happens if they do guess correctly? We end up passing through all the ifs without printing anything but once we return to the condition check for the while loop, we find that the condition is now false. We break out of the loop and print the congratulations statement.
```

Comment:
> Let’s assume that to_be_guessed has been set to some random number outside of this block of code.

I think the code block is hard to follow for a beginner as is, let's just set it to a number in the code instead of writing this sentence.

Comments line by line in the code would also really help readability. Ex:

```python
guess = int(input("New number: ")) # get a guess from the user
```